### PR TITLE
[onert/acl_common] Merge BackendContext

### PR DESCRIPTION
**Pull request by hseok-oh**
This commit introducees AclBackendContext to merge acl-cl and acl-neon BackendContext.

Signed-off-by: Hyeongseok Oh <hseok82.oh@samsung.com>

---

Related issue: https://github.com/Samsung/ONE/issues/8786

### CODE REVIEW

File: runtime/onert/backend/acl_common/AclBackendContext.h [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+{
+namespace acl_common
+{
+
+// TODO Find better way to handle common code (reduce template)
+template <typename T_TensorBuilder, typename T_ConstantInitializer, typename T_KernelGenerator,
+          typename T_Optimizer>
+class AclBackendContext : public onert::backend::BackendContext
+{
+public:
```
Comment (hseok-oh) [inline runtime/onert/backend/acl_common/AclBackendContext.h:38]: Below code is copy from `acl_cl` and `acl_neon`'s `BackendContext.cc`.

**Comment by hseok-oh** [comment]
@Samsung/one_onert PTAL

**Review by chunseoklee — APPROVED** [review]

**Review by jyoungyun — APPROVED** [review]
LGTM (+0.5)

**Comment by glistening** [comment]
(out of curiosity) @hseok-oh Does this PR improve duplicated code score?

**Review by glistening — APPROVED** [review]
LGTM

**Comment by chunseoklee** [comment]
> (out of curiosity) @hseok-oh Does this PR improve duplicated code score?

Yes

**Comment by glistening** [comment]
I asked because this PR moves duplicated code from `*.cpp` to `*.h`. But it seems to be same in term of compilation unit because two compilation units's line number would be same. I think dc score is measured based in term of code maintenance, not of object code duplication.

**Pull request merged**